### PR TITLE
find eigen using config mode

### PR DESCRIPTION
**Pull request by briansoe66**
* target_link_libraries eigen target, instead of including absolute path to eigen headers.
* find_dependency eigen in ifopt-config.cmake, because eigen is a public dependency
* remove FindEigen3.cmake and FindGTest.cmake modules. Both libraries support finding package via cmake config mode
* this change makes ifopt install files transferable to different filesystem location, because the absolute path to eigen headers is no longer hard-coded in the installed target files

### CODE REVIEW

File: ifopt-config.cmake [hunk 1]
```diff
@@ -35,6 +35,7 @@
 #
 #=============================================================================
 
+find_dependency(Eigen3 REQUIRED)
```
Comment (Levi-Armstrong) [inline ifopt-config.cmake:38]: It is my understanding that REQUIRED can be removed

**Review by Levi-Armstrong — CHANGES_REQUESTED** [review]

File: ifopt-config.cmake (hunk 1 above)
Comment (Levi-Armstrong) [reply ifopt-config.cmake:38]: See [here](https://cmake.org/cmake/help/latest/module/CMakeFindDependencyMacro.html) it should forward what was used when calling find_package.

Comment (briansoe66) [reply ifopt-config.cmake]: i've removed REQUIRED

**Comment by Levi-Armstrong** [comment]
@awinkler I just wanted to check to make sure this not going to impact other packages; you see any issue with merging this?

**Pull request merged**